lib/Elevate/Components/ELS.pm

```diff
@@ -0,0 +1,53 @@
+package Elevate::Components::ELS;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Components::ELS
+
+Remove ELS for CentOS 7
+
+=cut
+
+use cPstrict;
+
+use Elevate::OS ();
+
+use Cpanel::Pkgr  ();
+use Log::Log4perl qw(:easy);
+
+use parent qw{Elevate::Components::Base};
+
+use constant ELS_PACKAGE => 'els-define';
+
+sub pre_leapp ($self) {
+
+    return unless Elevate::OS::remove_els();
+
+    return unless Cpanel::Pkgr::is_installed(ELS_PACKAGE);
+
+    my @files_to_remove = qw{
+      /etc/yum/vars/elstoken
+      /etc/yum.repos.d/centos7-els.repo
+      /etc/yum.repos.d/centos7-els-rollout.repo
+    };
+
+    foreach my $file (@files_to_remove) {
+        if ( -e $file ) {
+            unlink $file or WARN("Could not remove file $file: $!");
+        }
+    }
+
+    $self->yum->remove(ELS_PACKAGE);
```

Comment:
This is my best guess too but I'm uncertain if there's more than one package to remove :\

Reply:
I got that from the manual steps outlined in the document referenced in the case.

Reply:
The rest is replacing system packages.  My guess is that leapp handles replacing the rest with upgrades for A8.

Reply:
agreed!